### PR TITLE
[opencc] Add new port

### LINKED ISSUE #7347: [New Port Request] OpenCC
Library name: OpenCC

Library description: A project for conversion between Traditional and Simplified Chinese

Source repository URL: https://github.com/BYVoid/OpenCC

Project homepage (if different from the source repository):

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

### PR DESCRIPTION
**Pull request by myd7349**
Fix #7347 .

### CODE REVIEW

**Comment by myd7349** [comment]
https://github.com/BYVoid/OpenCC/pull/350

File: ports/opencc/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        ${CURRENT_PACKAGES_DIR}/bin
+        ${CURRENT_PACKAGES_DIR}/debug/bin)
+else()
+    foreach(opencc_tool opencc opencc_dict opencc_phrase_extract)
+        file(REMOVE
+            ${CURRENT_PACKAGES_DIR}/bin/${opencc_tool}${VCPKG_TARGET_EXECUTABLE_SUFFIX}
+            ${CURRENT_PACKAGES_DIR}/debug/bin/${opencc_tool}${VCPKG_TARGET_EXECUTABLE_SUFFIX}
+        )
+    endforeach()
+endif()
```
Comment (vicroms) [inline ports/opencc/portfile.cmake]: This `else()` block should be part of the `if(tools IN_LIST FEATURES)`, as it is right now the produced tools will only be deleted when we are building the dynamic configuration.

Comment (vicroms) [inline ports/opencc/portfile.cmake]: This `else()` block should be part of the `if(tools IN_LIST FEATURES)`, as it is right now the produced tools will only be deleted when we are building the dynamic configuration.

**Review by vicroms — CHANGES_REQUESTED** [review]

File: ports/opencc/portfile.cmake (hunk 1 above)
Comment (myd7349) [reply ports/opencc/portfile.cmake]: Hi! @vicroms With a static built:
```cmake
    file(REMOVE_RECURSE
        ${CURRENT_PACKAGES_DIR}/bin
        ${CURRENT_PACKAGES_DIR}/debug/bin
    )
```
will delete those three files, and then the `bin` directory.
OpenCC will always build those three executables. So even installing without `tools` feature, the clean work is still necessary.

File: ports/opencc/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,51 @@
+include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/opencc/portfile.cmake]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/opencc/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            ${CURRENT_PACKAGES_DIR}/bin/${opencc_tool}${VCPKG_TARGET_EXECUTABLE_SUFFIX}
+            ${CURRENT_PACKAGES_DIR}/debug/bin/${opencc_tool}${VCPKG_TARGET_EXECUTABLE_SUFFIX}
+        )
+    endforeach()
+endif()
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include ${CURRENT_PACKAGES_DIR}/debug/share)
+
+# Handle copyright
+configure_file(${SOURCE_PATH}/LICENSE ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright COPYONLY)
```
Comment (LilyWangL) [inline ports/opencc/portfile.cmake]: Can you please use command `file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
Seems build failure on arm/uwp, can you check that?

Thanks.

**Comment by LilyWangL** [comment]
Pinging @myd7349 for response. Is work still being done for this PR?

**Review by LilyWangL — APPROVED** [review]

**Comment by strega-nil** [comment]
Alright, this looks good! Thanks @myd7349 :)

**Pull request merged**